src/Master.cxx

```diff
@@ -15,7 +17,19 @@ using std::string;
 
 std::string &CmdMsg_K::str() { return _str; }
 
-Master::Master(MainOpt &config) : config(config), command_listener(config) {}
+Master::Master(MainOpt &config) : config(config), command_listener(config) {
+  std::vector<char> buffer;
+  buffer.resize(128);
+  gethostname(buffer.data(), buffer.size());
+  if (buffer.back() != 0) {
+    // likely an error
+    buffer.back() = 0;
```

Comment:
The buffer size is chosen such that every possible hostname should fit in. Problem is that it is not trivial to find the required size of the buffer. If for some reason it didn't fit, then there is anyway no way to find out whether only the null terminator got left out. So we can as well just overwrite the last byte. This is really just for safety. I can add a runtime assert and log...